lib/mayaUsd/ufe/ProxyShapeSceneSegmentHandler.cpp

```diff
@@ -45,6 +45,41 @@ ProxyShapeSceneSegmentHandler::create(const Ufe::SceneSegmentHandler::Ptr& mayaS
 // Ufe::SceneSegmentHandler overrides
 //------------------------------------------------------------------------------
 
+#if (UFE_PREVIEW_VERSION_NUM >= 4033)
+Ufe::Selection
+ProxyShapeSceneSegmentHandler::findGatewayItems_(const Ufe::Path& path, Ufe::Rtid nestedRtid) const
+{
+    Ufe::Selection result = Ufe::Selection();
+
+    // Handle other gateway node types that MayaUSD is not aware of.
+    // `nestedRtid` is used as a filter. If it matches the MayaUSD runtime ID, MayaUSD is aware of
+    // the requested gateway items so this step can be skipped.
+    if (nestedRtid != g_USDRtid && fMayaSceneSegmentHandler) {
+        result = fMayaSceneSegmentHandler->findGatewayItems_(path, nestedRtid);
+    }
+
+    // Find the MayaUSD proxyShapes.
+    // `nestedRtid` is used as a filter. Only add MayaUSD proxyShapes to the result if the argument
+    // matches the MayaUSD runtime ID or if the argument is 0 which means return all.
+    if (nestedRtid == 0 || nestedRtid == g_USDRtid) {
```

Comment:
Zero does not feel magic enough as it is the RtID of the first registered handler as seen in  `Ufe::RunTimeMgr::register_()`. Can you explictly declare a `Ufe::RuntimeMgr::AllRtID` constant with value `std::numeric_limits<RtID>::max` and use that instead?

Reply:
As far as I understood it, `0` is an invalid runtime ID. This is also indicated by [this comment](https://git.autodesk.com/media-and-entertainment/ufe/blob/3624d2f225abd3044fcba31cfa6c4760deda92fe/src/runTimeMgr.cpp#L76). 

I think this is why `0` was used in the [proposed interface](https://git.autodesk.com/media-and-entertainment/ufe/pull/297). It may still make sense to add a constant like `Ufe::RuntimeMgr::AllRtid` or `Ufe::RuntimeMgr::InvlaidRtid` but I feel like the value itself should be fine.

Reply:
Agree, zero is the correct value, but please still introduce it as a constant. Thanks.

Reply:
The Ufe rtid.h does indeed have a comment stating that "ID 0 is used to represent an invalid runtime ID." - its too bad it didn't also include a constant for that. 

BTW, this is the open source github repo so we should refrain from using links to our internal repos.

Reply:
It looks like the upper section of this method (in the new #if) is new and the bottom part is common to both if/else. If so can we just have the if inside this method for new code?

Reply:
Who is calling the new version with the extra Rtid param?

Reply:
The new version of the method will be used by LookdevX to get a list of gateways into USD. 

Having the `#if`/`#else` inside of the method would be great but I don't think we can, can we? The new version changes the signature of the method, so the signature has to be part of the `#if`/`#else`. Most of the code inside of the method is touched as well (i.e., enclosed in some comparison with `nestedRtid`). The only thing that remains completely unchanged is the comment about `If there were Usd prims that are gateway items` and the return statement.

We could have something like this, but I felt like it would be less readable:
```C++
#if (UFE_PREVIEW_VERSION_NUM >= 4033)
Ufe::Selection ProxyShapeSceneSegmentHandler::findGatewayItems_(const Ufe::Path& path, Ufe::Rtid nestedRtid) const
{
    // new code
#else
Ufe::Selection ProxyShapeSceneSegmentHandler::findGatewayItems_(const Ufe::Path& path)
{
    // old code
#endif
    // common code
}
```
Let me know what you think. Maybe there is another way to do it more elegantly that I'm not aware of.

Reply:
I think based on the latest changes request to ufe, this will become an overloaded method with the new Rtid param.